lib/src/pipeline/aggregation/online_tdigest.rs

```diff
@@ -29,7 +29,22 @@ pub struct OnlineTdigest {
     state: Mutex<State>,
 }
 
-#[derive(Default, Debug)]
+impl Clone for OnlineTdigest {
+    fn clone(&self) -> Self {
+        match self.state.try_lock() {
+            Ok(state) => {
+                Self {
+                    state: Mutex::new(state.clone())
+                }
+            },
+            Err(_) => {
+                OnlineTdigest::default()
+            },
+        }
+    }
```

Comment:
goodmetrics avoids silently failing and silently propagating lies. This should never happen, but if the mutex is poisoned then there is a correctness bug _or_ an unrecoverable environment issue, like oom.

```suggestion
    fn clone(&self) -> Self {
        Self {
            state: Mutex::new(self.state.lock().expect("lock should never fail").clone())
        }
    }
```